overlord/configstate/config/helpers.go

```diff
@@ -135,6 +135,37 @@ func GetFromChange(snapName string, subkeys []string, pos int, config map[string
 	return GetFromChange(snapName, subkeys, pos+1, configm, result)
 }
 
+// GetSnapConfig retrieves the raw configuration of a given snap.
+func GetSnapConfig(st *state.State, snapName string) (json.RawMessage, error) {
+	var config map[string]*json.RawMessage
+	err := st.Get("config", &config)
+	if err == state.ErrNoState {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	snapcfg, ok := config[snapName]
+	if !ok {
+		return nil, nil
+	}
+	return *snapcfg, nil
+}
+
+// SetSnapConfig replaces the configuration of a given snap.
+func SetSnapConfig(st *state.State, snapName string, snapcfg json.RawMessage) error {
```

Comment:
for symmetry with other Set  we probably want to take a *  and when nil, delete instead of setting in the map